gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -223,6 +224,15 @@ protected void addLineageSourceInfo(SourceState sourceState, SourceEntity entity
       }
     } catch (Exception e) {
       throw new RuntimeException(e);
+    } finally {
+      if(salesforceExtractor != null) {
+        try {
+          // Only close connection here since we don't want to update the high watermark for the workUnitState here
```

Comment:
I specifically call closeConnection instead close method is because in close method, we first set the high watermark for the workunitState and then call closeConnection. Although I see we create a new workunitState when creating the extractor and setting HWM for that should not affect the correctness, I want to avoid any risk in the future and only call closeConnection to close the resources

Reply:
makes sense.  so maybe?
```
// `.closeConnection()`, not `.close()` to avoid updating high watermark for the workUnitState
```